Source/DataModels/Components/ComponentMockState.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include "Component.h"
+
+#include "Math/float3.h"
+
+class ComponentMockState :
+	public Component
+{
+public:
+	ComponentMockState(bool active, GameObject* owner);
+	~ComponentMockState() override;
+
+	void Update() override;
+	void Draw() override;
+
+	void SaveOptions(Json& meta) override;
+	void LoadOptions(Json& meta) override;
+
+	bool GetIsWinState();
+	bool GetIsFailState();
+
+	void SetIsWinState(bool isWinState);
+	void SetIsFailState(bool isFailState);
+
+private:
+
+	bool isWinState;
+	bool isFailState;
+};
+
+inline bool ComponentMockState::GetIsWinState()
```

Comment:
All getters should be const
```suggestion
inline bool ComponentMockState::GetIsWinState() const
```
Same for the rest.